### PR TITLE
fix bug for node metrics on overview page

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <talonzhu@yunify.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubesphere/kubesphere/issues/3714

**Special notes for reviewers**:
```
I will add a query parameter `queryType` in the back-end to handle the query parameter `type`.
```

**Additional documentation, usage docs, etc.**:

### CODE REVIEW

File: pkg/kapis/monitoring/v1alpha3/register.go [hunk 1]
```diff
@@ -80,6 +80,7 @@ func AddToContainer(c *restful.Container, k8sClient kubernetes.Interface, monito
 		Param(ws.QueryParameter("sort_type", "Sort order. One of asc, desc.").DefaultValue("desc.").DataType("string").Required(false)).
 		Param(ws.QueryParameter("page", "The page number. This field paginates result data of each metric, then returns a specific page. For example, setting **page** to 2 returns the second page. It only applies to sorted metric data.").DataType("integer").Required(false)).
 		Param(ws.QueryParameter("limit", "Page size, the maximum number of results in a single page. Defaults to 5.").DataType("integer").Required(false).DefaultValue("5")).
+		Param(ws.QueryParameter("type", "The query type. This field can be 'rank' or ''").DataType("string").Required(false).DefaultValue("")).
```
Comment (benjaminhuo) [inline pkg/kapis/monitoring/v1alpha3/register.go:83]: The query type. This field can be 'rank' or ''
=>
The query type. This field can be set to 'rank' for node ranking query or '' for others. Defaults to ''.

File: pkg/models/monitoring/monitoring.go [hunk 2]
```diff
@@ -115,16 +115,23 @@ func (mo monitoringOperator) GetMetricOverTime(expr, namespace string, start, en
 func (mo monitoringOperator) GetNamedMetrics(metrics []string, time time.Time, opt monitoring.QueryOption) Metrics {
 	ress := mo.prometheus.GetNamedMetrics(metrics, time, opt)
 
-	if mo.metricsserver != nil {
+	opts := &monitoring.QueryOptions{}
+	opt.Apply(opts)
+
+	var isNodeRankActivated bool
```
Comment (benjaminhuo) [inline pkg/models/monitoring/monitoring.go:121]: isNodeRankActivated
=>
isNodeRankingQuery

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: pkg/models/monitoring/monitoring.go (hunk 2 above)
Comment (zhu733756) [reply pkg/models/monitoring/monitoring.go]: fixed

File: pkg/kapis/monitoring/v1alpha3/register.go (hunk 1 above)
Comment (zhu733756) [reply pkg/kapis/monitoring/v1alpha3/register.go]: fixed

**Review by benjaminhuo — APPROVED** [review]
/lgtm

**Comment by benjaminhuo** [comment]
@zhu733756 Is there any issue opened for this ? Pls link a issue to this.

/assign @zryfish

**Comment by zhu733756** [comment]
> @zhu733756 Is there any issue opened for this ? Pls link a issue to this.
> 
> /assign @zryfish

https://github.com/kubesphere/kubesphere/issues/3714

**Review by benjaminhuo — APPROVED** [review]
/lgtm

**Pull request merged**